llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -286,6 +300,12 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineIntrinsic(
   case Intrinsic::aie2_vinsert8_I512: {
```

Comment:
I was wondering if we can refactor in some way this scalar size information. We use this information here and in `IsVInsert`. If we have something like `getVInsertScalarSize(...)` that returns zero if the intrinsic is not know would help. For example, in your previous lambda will be enough use `return getVInsertScalarSize(cast<GIntrinsic>(*MI).getIntrinsicID()) == SclSrcBits;` You can also drop your switch-case here and pass directly the value of `getVInsertScalarSize(...)` to `tryToCombineVectorInserts` if it is greater than 0.